### PR TITLE
feat: add initial qwen2.5-vl model and test

### PR DESCRIPTION
**Pull request by drbh**
This PR adds support for qwen2.5-vl models and currently loads the weights and supports reasonable responses. Opening early for exposure and any feedback.

These changes are dependent on https://github.com/huggingface/text-generation-inference/pull/2943 and must be rebased/merged after it is merged

## items 
- [x] allow qwen2_5_vl models to load in server/router
- [x] add updated vision model 
  - [x] adjust weight names/loading
  - [x] adjust config naming
  - [x] `RMSNorm` in vision blocks
  - [x] update MLP for up/gate/down
  - [x] add `fullatt_block_indexes` 
- [x] validation
  - [x] add tests
  - [x] add snapshots
  - [x] tests more outputs locally for edge cases
  - [x] inspect logits for correctness
  - [x] ensure mrope is producing correct ids

small reproducible example:

```bash
text-generation-launcher --model-id Qwen/Qwen2.5-VL-3B-Instruct
```

script 
```python
import requests
import json

url = "http://127.0.0.1:3000/generate"
headers = {"Content-Type": "application/json"}
image_urls = [
    "https://cdn.britannica.com/61/93061-050-99147DCE/Statue-of-Liberty-Island-New-York-Bay.jpg",
    "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/tgi/rabbit.png",
]

for image in image_urls:
    query = "Describe the image"
    payload = {
        "inputs": f"<|im_start|>system\nYou are a helpful assistant.<|im_end|>\n<|im_start|>user\n![]({image}){query}<|im_end|>\n<|im_start|>assistant\n",
        "parameters": {"max_new_tokens": 50},
    }
    response = requests.post(url, headers=headers, json=payload)
    print(json.dumps(response.json(), indent=4))

```

output
```json
{
    "generated_text": "The image showcases the iconic Statue of Liberty in New York City, with the New York City skyline in the background. The statue is a large, green-colored sculpture on a stone pedestal, with the American flag on a flagpole in the foreground. The"
}
{
    "generated_text": "The image depicts a character in a space suit, set in a rocky, desert-like environment with a warm, orange hue. The character is a large, brown, and white rabbit with long, pointed ears and a small, beak-like nose."
}
```

### CODE REVIEW

**Review by Narsil — APPROVED** [review]
LGTM!

File: nix/overlay.nix [hunk 1]
```diff
@@ -13,16 +13,16 @@ final: prev: {
...
+        transformers = python-super.transformers.overrideAttrs (
+          _: _: {
+            src = final.fetchFromGitHub {
+              owner = "huggingface";
+              repo = "transformers";
+              rev = "8d73a38606bc342b370afe1f42718b4828d95aaa";
+              hash = "sha256-MxroG6CWqrcmRS+eFt7Ej87TDOInN15aRPBUcaycKTI=";
+            };
+          }
+        );
```
Comment (Narsil) [inline nix/overlay.nix:25]: Do we still need that with 4.49 release ?

File: server/text_generation_server/models/custom_modeling/qwen2_5_vl.py [hunk 2]
```diff
@@ -0,0 +1,974 @@
...
+    videos_kwargs: Qwen2_5_VLVideosProcessorKwargs
+    _defaults = {
+        "text_kwargs": {
+            "padding": False,
+        },
+        "videos_kwargs": {"fps": 2.0},
+    }
+
+
+class Qwen2_5_VLProcessor(ProcessorMixin):
```
Comment (Narsil) [inline server/text_generation_server/models/custom_modeling/qwen2_5_vl.py:77]: Is this not defined in `transformers` ?

File: nix/overlay.nix (hunk 1 above)
Comment (drbh) [reply nix/overlay.nix:25]: this should be able to be removed with 4.49

File: server/text_generation_server/models/vlm_causal_lm.py [hunk 3]
```diff
@@ -422,7 +430,10 @@ def forward(
             max_s = batch.max_current_length
             lm_head_indices = batch.prefill_head_indices
 
-        if self.model.config.model_type == "qwen2_vl":
+        if (
+            self.model.config.model_type == "qwen2_vl"
+            or self.model.config.model_type == "qwen2_5_vl"
+        ):
```
Comment (Narsil) [inline server/text_generation_server/models/vlm_causal_lm.py:436]:
```suggestion
        if self.model.config.model_type in {"qwen2_vl", "qwen2_5_vl"}:
```
Fixing the chatbot for you :)

File: server/text_generation_server/models/vlm_causal_lm.py [hunk 4]
```diff
@@ -231,7 +236,10 @@ def batch_tokenized_inputs(
                     image = Image.open(BytesIO(chunk.image.data))
                     # qwen2_vl expects images to be greater than 20 pixels, this is for warmup since the
                     # default warmup image is 20x20
-                    if config.model_type == "qwen2_vl":
+                    if (
+                        config.model_type == "qwen2_vl"
+                        or config.model_type == "qwen2_5_vl"
+                    ):
```
Comment (Narsil) [inline server/text_generation_server/models/vlm_causal_lm.py:242]: NIT: Simpler condition.

File: server/text_generation_server/models/custom_modeling/qwen2_5_vl.py (hunk 2 above)
Comment (drbh) [reply server/text_generation_server/models/custom_modeling/qwen2_5_vl.py:77]: this was added in 4.49 so can be be removed with the update too

**Review by Narsil — APPROVED** [review]
